app/src/main/java/com/github/se/eduverse/viewmodel/PhotoViewModel.kt

```diff
@@ -20,10 +25,23 @@ open class PhotoViewModel(private val photoRepository: IPhotoRepository) : ViewM
   open val photos: StateFlow<List<Photo>>
     get() = _photos
 
-  fun savePhoto(photo: Photo) {
+  fun savePhoto(
+      photo: Photo,
+      folder: Folder? = null,
+      addToFolder: (String, String, Folder) -> Unit = { _, _, _ -> }
+  ) {
     viewModelScope.launch {
       val success = photoRepository.savePhoto(photo)
       _savePhotoState.value = success
+      if (folder != null) {
```

Comment:
I think that it's better to define another function savePhotoToFolder that is called after the user selects the folder from the bottom sheet dialog and calls itself photoRepository.savePhoto it results in a better user flow since the user doesn't need to click on save after selecting a folder, as it's implicitly intended that the user chooses to save the picture to cloud storage when he clicks on add to folder

Reply:
This would be a good behavior in theory, but the problem is the photo doesn't exist on the firebase storage until the user calls save. So if we add the photo to the folder immediatly but then decide to cancel (either by pressing the cross or by closing the app), then it would create problem.

Reply:
But I think that after clicking on add to folder and selecting the folder the photo should be saved to firebase storage and added to folder and the user is navigated out of the next screen, this way he can't cancel

Reply:
Done